src/main/java/longtimenosee/model/event/Date.java

```diff
@@ -34,6 +35,23 @@ public Date(String value) {
         this.date = parseDate(value);
     }
 
+    /**
+     * Requires a dateTime object to already have an appropriate format (YYYY-MM-DD).
+     * Checks if given Year, Month, Date values are integer values in the appropriate range
+     * (e.g: 0 <= MM <= 12)
+     * @param value in YYYY-MM-DD.
+     * @return
+     */
+    public static boolean isValidRange(String value) {
+        assert(isValidFormat(value) == true);
```

Comment:
Good use of assert statements as a failsafe!

Reply:
Actually I realized I could just do `assert(isValidFormat(value)`, bc what's inside assert is a condition!